Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs

```diff
@@ -278,9 +278,9 @@ public ActionResult BulkDeleteSubmissions([DataSourceRequest]DataSourceRequest r
 
         public JsonResult GetSubmissionTypes(int problemId, bool? allowBinaryFilesUpload)
         {
-            var selectedProblemContest = this.Data.Contests.All().FirstOrDefault(contest => contest.Problems.Any(problem => problem.Id == problemId));
+            var selectedProblem = this.Data.Problems.All().FirstOrDefault(problem => problem.Id == problemId);
```

Comment:
Why you get the problem with one query, and then you got all problem submissionTypes with another?
You can combine this row and the row below.